stats/stats/src/data_source/kinds/remote_db/query/one.rs

```diff
@@ -45,10 +47,136 @@ where
     ) -> Result<TimespanValue<Resolution, Value>, ChartError> {
         let query = S::get_statement(&cx.blockscout_applied_migrations);
         let data = TimespanValue::<Resolution, Value>::find_by_statement(query)
-            .one(cx.blockscout.connection.as_ref())
+            .one(cx.blockscout)
             .await
             .map_err(ChartError::BlockscoutDB)?
             .ok_or_else(|| ChartError::Internal("query returned nothing".into()))?;
         Ok(data)
     }
 }
+
+/// Will reuse result for the same produced query within one update
+/// (based on update context)
+pub struct PullOne24hCached<S, Value>(PhantomData<(S, Value)>)
+where
+    S: StatementFromRange,
+    Value: FromQueryResult + Cacheable + Clone + Send;
+
+impl<S, Value> RemoteQueryBehaviour for PullOne24hCached<S, Value>
+where
+    S: StatementFromRange,
+    Value: FromQueryResult + Cacheable + Clone + Send,
+{
+    type Output = TimespanValue<NaiveDate, Value>;
+
+    async fn query_data(
+        cx: &UpdateContext<'_>,
+        _range: UniversalRange<DateTime<Utc>>,
+    ) -> Result<TimespanValue<NaiveDate, Value>, ChartError> {
+        let update_time = cx.time;
+        let range_24h = update_time
+            .checked_sub_signed(TimeDelta::hours(24))
+            .unwrap_or(DateTime::<Utc>::MIN_UTC)..=update_time;
+        let query = S::get_statement(
+            Some(inclusive_range_to_exclusive(range_24h)),
+            &cx.blockscout_applied_migrations,
+        );
+
+        let value = if let Some(cached) = cx.cache.get::<Value>(&query).await {
+            cached
+        } else {
+            let find_by_statement = Value::find_by_statement(query.clone());
+            let value = find_by_statement
+                .one(cx.blockscout)
+                .await
+                .map_err(ChartError::BlockscoutDB)?
+                .ok_or_else(|| ChartError::Internal("query returned nothing".into()))?;
+            cx.cache.insert(&query, value.clone()).await;
```

Comment:
never remove from cache => potential memory leak?

Reply:
suggest to use `SizedCache` from cache crate

Reply:
no, cache is created each from scratch each time there is an update.
all arc objects should also be destroyed after each update together with `UpdateContext`.
note that cache is in update context, so it's a cache specifically for this update.

Reply:
double-checked, it indeed works like that.
`from_params_now_or_override` creates a new cache.

https://github.com/blockscout/blockscout-rs/blob/77266de8cea27ba3742f3679f2f6b72ba539fcb2/stats/stats/src/update_group.rs#L329-L342

Reply:
added comment to clarify